File: 동양미래대학교 고급알고리즘/2주차/CoinCalc.py
cost = int(input("비용 입력>"))
n = int(input("반복 횟수>"))
coin = []
cnt = 0

for i in range(n):
    cn = int(input("동전입력>"))
    coin.append(cn)

for i in coin:
    cnt += cost//i
    print(i, "원 동전 개수>", int(cost/i))
    cost %= i

print("필요한 동전의 개수>", cnt)

File: dongyang_algo/day08/ch2/CoinCalc.java
package day08.ch2;

// 시간 복잡도 O(N)
public class CoinCalc {
    public static void main(String[] args) {
        int charge = 4200;
        int cnt = 0;
        int[] coin = { 50000, 10000, 5000, 1000, 500, 100, 50, 10, 5, 1 };

        for (int i = 0; i < coin.length; i++) {
            if (charge == 0)
                break;
            cnt += charge / coin[i];
            charge %= coin[i];
        }
        System.out.println("총 개수>" + cnt);
    }
}